contracts/V1/StateChannelDiamondProxy/AStateChannelManagerProxy.sol

```diff
@@ -227,6 +113,78 @@ abstract contract AStateChannelManagerProxy is StateChannelManagerInterface, Sta
         );
     }
 
+    function updateStateSnapshotFork(
+        bytes32 channelId,
+        StateSnapshot memory newStateSnapshot,
+        ExitChannelBlock[] memory exitChannelBlocks
+    ) public override {
+        _delegatecall(
+            address(stateSnapshotFacet),
+            abi.encodeCall(stateSnapshotFacet.updateStateSnapshotFork, (channelId, newStateSnapshot, exitChannelBlocks))
+        );
+    }
+
+    function updateStateSnapshotSameFork(
+        bytes32 channelId,
+        MilestoneProof[] memory milestoneProofs,
+        StateSnapshot[] memory milestoneSnapshots,
+        ExitChannelBlock[] memory exitChannelBlocks
+    ) public override {
+        _delegatecall(
+            address(stateSnapshotFacet),
+            abi.encodeCall(
+                stateSnapshotFacet.updateStateSnapshotSameFork,
+                (channelId, milestoneProofs, milestoneSnapshots, exitChannelBlocks)
+            )
+        );
+    }
+
+    function joinChannel(JoinChannelConfirmation memory joinChannelConfirmations) public override {
+        _delegatecall(
+            address(joinChannelFacet), abi.encodeCall(joinChannelFacet.joinChannel, (joinChannelConfirmations))
+        );
+    }
+
+    // ********** public/external DIAMOND functions **********
+
+    /// @dev Callable only by diamond facets - performs the deposit of the specific assets by interpeting `joinChannel` - returns bool success
+    function depositAssetsComposable(JoinChannel memory joinChannel) public onlySelf returns (bool) {
+        return _depositAssetsComposable(joinChannel);
+    }
+
+    /// @dev Callable only by diamond facets - performs the withdrawal of the specific assets by interpeting `exitChannel` - returns bool success
+    function withdrawAssetsComposable(ExitChannel memory exitChannel) public onlySelf returns (bool) {
+        return _withdrawAssetsComposable(exitChannel);
+    }
+
+    function applySlashesToStateMachine(bytes memory encodedState, address[] memory slashedParticipants)
+        public
+        onlySelf
+        returns (bytes memory encodedModifiedState, ExitChannel[] memory)
+    {
+        return _applySlashesToStateMachine(encodedState, slashedParticipants);
+    }
+
+    function removeParticipantsFromStateMachine(bytes memory encodedState, address[] memory participants)
+        public
+        onlySelf
+        returns (bytes memory encodedModifiedState, ExitChannel[] memory)
+    {
+        return _removeParticipantsFromStateMachine(encodedState, participants);
+    }
+
+    function executeStateTransitionOnState(bytes32 channelId, bytes memory encodedState, Transaction memory _tx)
```

Comment:
state transition is always on state and it's clear enough from the function signature


```suggestion
    function executeStateTransition(bytes32 channelId, bytes memory encodedState, Transaction memory _tx)
```